### PR TITLE
Fix hdmaster-key / seed-key confusion

### PR DESCRIPTION
**Pull request by jonasschnelli**
Addresses #12084 and #8684

This renames a couple of functions and members (no functional changes, expect log prints):
* Rename CKey::SetMaster to CKey::SetSeed
* Rename CHDChain::masterKeyId to CHDChain::seedKeyID
* Rename CHDChain::hdMasterKeyID to CHDChain::hdSeedKeyID
* Rename CWallet::GenerateNewHDMasterKey to CWallet::GenerateNewHDSeedKey
* Rename CWallet::SetHDMasterKey to CWallet::SetHDSeedKey

As well it introduces a tiny API change (a649ab2):
* RPC API change: Rename "hdmasterkeyid" to "hdseedkeyid", rename "hdmaster" in wallet-dump output to "hdseed"

Fixes also a bug (16d49e1):
* Bugfix: use "s" instead of the incorrect "m" for the seed-key hd-keypath key metadata

### CODE REVIEW

**Comment by achow101** [comment]
utACK 16d49e1b82808803d6c492056219e53791d14cfe

Couldn't you have done this with scripted diffs?

It would be nice to include the actual `hdmasterkeyid` in the `getwalletinfo` output, not just the seed id.

File: src/wallet/rpcdump.cpp [hunk 1]
```diff
@@ -733,14 +733,14 @@ UniValue dumpwallet(const JSONRPCRequest& request)
     file << strprintf("#   mined on %s\n", EncodeDumpTime(chainActive.Tip()->GetBlockTime()));
     file << "\n";
 
-    // add the base58check encoded extended master if the wallet uses HD
-    CKeyID masterKeyID = pwallet->GetHDChain().masterKeyID;
-    if (!masterKeyID.IsNull())
+    // add the base58check encoded extended master if the wallet uses HD
```
Comment (MarcoFalke) [inline src/wallet/rpcdump.cpp]: hidden whitespace change :eye:

Comment (jonasschnelli) [reply src/wallet/rpcdump.cpp]: Oh! Thanks. Fixed.

**Comment by MarcoFalke** [comment]
Would be nice to do this as scripted diff

**Comment by jnewbery** [comment]
Implemented as a scripted diff here: https://github.com/jnewbery/bitcoin/tree/master_key_to_seed

@jonasschnelli - are you still interested in maintaining this? If not, I can open a new PR.

**Comment by jnewbery** [comment]
I've opened #12924, which is this PR reimplemented as a scripted diff.

@jonasschnelli - I'm very happy to close that PR if you want to start maintaining this one again. Just let me know!

**Comment by JenniferBrooks20132017LSTSEMDAJJPBN** [comment]
Jj B has sent you an email via Gmail confidential mode:

[image: Gmail logo]found finally just seeing where it goes someone has to stop stealing what i pay for my buissness <https://confidential-mail.google.com/msg/AA12eCjyOYqJ5G4K4Nv24Fv9rXkJ99YNf0FmM_tgyFh_f4qDH73-acz-XeMFsfH0KLBcQDzVaHNMNdsKyBZeXw8wXSsELTOTz1Hn6q-TisNbm6xn4cfQ1ikkz9SmD4q4O6kdu0iNvJDkFs1LskrxlZM3wZgJekyOCmd-Hy32DTO7B9qQI4IZrsnwO_o0PpvZUx0DN9slM7fUq3GvXIniHIbPdW1vAF5osOieTFe7GfSuFrLw2I1yhqHFZCZ91XTyr0qcgWAl>

This message was sent on Mar 31, 2020 at 7:13:45 PM PDT
You can open it by clicking the link below. This link will only work for reply@reply.github.com.

View the email <https://confidential-mail.google.com/msg/AA12eCjyOYqJ5G4K4Nv24Fv9rXkJ99YNf0FmM_tgyFh_f4qDH73-acz-XeMFsfH0KLBcQDzVaHNMNdsKyBZeXw8wXSsELTOTz1Hn6q-TisNbm6xn4cfQ1ikkz9SmD4q4O6kdu0iNvJDkFs1LskrxlZM3wZgJekyOCmd-Hy32DTO7B9qQI4IZrsnwO_o0PpvZUx0DN9slM7fUq3GvXIniHIbPdW1vAF5osOieTFe7GfSuFrLw2I1yhqHFZCZ91XTyr0qcgWAl>

Gmail confidential mode gives you more control over the messages you send. The sender may have chosen to set an expiration time, disable printing or forwarding, or track access to this message. Learn more <https://support.google.com/mail/answer/7674059>
Gmail: Email by Google
Use is subject to the Google Privacy Policy <https://myaccount.google.com/privacypolicy?hl=en>
Google LLC, 1600 Amphitheatre Parkway, Mountain View, CA 94043, USA
You have received this message because someone sent you an email via Gmail confidential mode.
[image: Google logo]

**Comment by JenniferBrooks20132017LSTSEMDAJJPBN** [comment]
Your Google verification code is 181127

**Pull request closed without merging**